app/policies/policy_context.rb

```diff
@@ -0,0 +1,12 @@
+class PolicyContext
+  attr_reader :record
+
+  def initialize(record, user)
+    @record = record
+    @user = user
+  end
+
+  def policy_class
+    "#{@user.roles.uniq.first}DefenceRequestPolicy".classify
```

Comment:
@novotnyjakub as discussed offline: The multiple role issue can probably be handled in the short term by finding the role with the 'highest' level of permissions, any other roles will then be a subset of those roles. With regards to the repetition of action names across controllers, agreed that most of the small controllers can be given a single policy clause 'close_dr' for instance which would cover the whole controller rather than individual actions and we assume that the standard CRUD actions relate to the defence_request_controller as the policies are named by the DR model, ie CcoDefenceRequestPolicy which is a side effect of the pundit implementation